src/hooks/encounter/encounter.ts

```diff
@@ -0,0 +1,58 @@
+import { useCallback, useState } from 'react';
+
+import { isUser } from '../../jsonValidation';
+import { User } from '../../types';
+
+const createEncounterAsync = async (
+  qrId: string,
+  userId: string
+): Promise<{ data?: User; error: boolean }> => {
+  // TODO: implement API Call
+  const res = {
+    birthDate: new Date(42).toISOString(),
+    createdAt: new Date(1337).toISOString(),
+    firstName: 'Max',
+    imageUri: 'https://smart-village.solutions/wp-content/uploads/2020/01/Services.png',
+    lastName: 'Mustermann',
+    phone: '0123 123123',
+    userId: 'userId',
+    verified: false,
+    village: 'Utopia'
+  };
+
+  const error = !isUser(res);
+
+  return {
+    data: error ? undefined : res,
+    error
+  };
+};
+
+export const useCreateEncounter = (
+  onSuccess: (user: User) => void,
+  onError: () => void
+): {
+  createEncounter: (qrId: string, userId: string) => void;
+  loading: boolean;
+} => {
+  const [loading, setLoading] = useState(false);
+
+  const createEncounter = useCallback(async (qrId: string, userId: string) => {
+    setLoading(true);
+    try {
+      const res = await createEncounterAsync(qrId, userId);
+
+      if (!res.data || res.error) {
```

Comment:
i think it will be necessary to check for the status code to decide wether there was success or an error

Reply:
i will revisit all error handling upon implementing the api calls.

i don't know whether i will include the error logic here, or already inside of the `createEncounterAsync` function.

Reply:
changed in #339 
`createEncounterAsync` will now return a parsed and validated result or nothing. so we only need to check for presence of `res` here. (the used data is no longer nested as `data` but is directly returned)